### PR TITLE
Fix bug comment on postgres and others management

### PR DESCRIPTION
**Pull request by Ailurupoda**
## Description
Fix issues created by postgres management on other data management system

## Checklist
- Use of connector and plugin file to launch query
- Define a supportsComment method to siwtch display comment if PGDatabase found
- Method commentTable(self, schema, tablename, comment=None) to comment/unComment a postgres Table
- Method getComment(self, tablename, field) to display fields comment

### CODE REVIEW

File: python/plugins/db_manager/db_plugins/spatialite/connector.py [hunk 1]
```diff
@@ -496,7 +494,6 @@ def renameTable(self, table, new_table):
             self._execute(c, sql)
 
         self._commit()
-        return True
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/spatialite/connector.py]: Are these changes on purpose? They look unrelated to this fix

File: python/plugins/db_manager/dlg_import_vector.py [hunk 2]
```diff
@@ -25,7 +25,8 @@
 from builtins import range
 
 from qgis.PyQt.QtCore import Qt, QFileInfo
-from qgis.PyQt.QtWidgets import QDialog, QFileDialog, QMessageBox
+from qgis.PyQt.QtWidgets import QDialog, QFileDialog, QMessageBox, QApplication
+from qgis.PyQt.QtGui import QCursor
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_import_vector.py]: What's this import for?

File: python/plugins/db_manager/dlg_import_vector.py [hunk 3]
```diff
@@ -25,7 +25,8 @@
 from builtins import range
 
 from qgis.PyQt.QtCore import Qt, QFileInfo
-from qgis.PyQt.QtWidgets import QDialog, QFileDialog, QMessageBox
+from qgis.PyQt.QtWidgets import QDialog, QFileDialog, QMessageBox, QApplication
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_import_vector.py]: What's QApplication used for?

File: python/plugins/db_manager/dlg_import_vector.py [hunk 4]
```diff
@@ -51,6 +52,11 @@ def __init__(self, inLayer, outDb, outUri, parent=None):
         self.outUri = outUri
         self.setupUi(self)
 
+        objClass = self.db.searchClass()
+        if objClass != "PGDatabase":
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_import_vector.py]: This should not have any knowledge about the database behind. Rather call a method that asks the database if it supports comments.

Like `self.db.supportComments()`

File: python/plugins/db_manager/dlg_import_vector.py [hunk 5]
```diff
@@ -368,9 +374,9 @@ def accept(self):
             self.db.connector.createSpatialIndex((schema, table), geom)
 
         # add comment on table
-        if self.chkCom.isEnabled() and self.chkCom.isChecked():
+        if self.chkCom.isEnabled() and self.chkCom.isChecked() and objClass == "PGDatabase":
             # using connector executing COMMENT ON TABLE query (with editCome.text() value)
-            self.db.connector._execute(None, 'COMMENT ON TABLE "{0}"."{1}" IS E\'{2}E\''.format(schema, table, self.editCom.text()))
+            self.db.connector._execute(None, 'COMMENT ON TABLE "{0}"."{1}" IS E\'{2}\''.format(schema, table, self.editCom.text()))
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_import_vector.py]: This needs to go into the provider plugin, it's postgres specific code

File: python/plugins/db_manager/dlg_table_properties.py [hunk 6]
```diff
@@ -50,6 +50,10 @@ def __init__(self, table, parent=None):
 
         self.db = self.table.database()
 
+        objClass = self.db.searchClass()
+        if objClass != "PGDatabase":
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_table_properties.py]: Same as above, call a method that determines if there's support for comments in here (and don't explicitly check provider types)

File: python/plugins/db_manager/dlg_table_properties.py [hunk 7]
```diff
@@ -333,37 +337,25 @@ def deleteIndex(self):
                 DlgDbError.showError(e, self)
 
     def createComment(self):
-        """Adds a comment to the selected table"""
-
+        #Function that add a comment to the selected table
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_table_properties.py]: Leave the comment, it's quite clear it's a function ;)

File: python/plugins/db_manager/dlg_table_properties.py [hunk 8]
```diff
@@ -333,37 +337,25 @@ def deleteIndex(self):
...
-            QMessageBox.information(self, self.tr("Add comment"), self.tr("Comments are not supported for this database."))
-            return
         self.refresh()
         #Display successful message
         QMessageBox.information(self, self.tr("Add comment"), self.tr("Table successfully commented"))
 
     def deleteComment(self):
-        """Drops the comment on the selected table"""
-
+        #Function that drop the comment to the selected table
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_table_properties.py]: Same, leave the old docstring

File: python/plugins/db_manager/dlg_field_properties.py [hunk 9]
```diff
@@ -42,37 +41,22 @@ def __init__(self, parent=None, fld=None, table=None, db=None):
...
-            # Check if result is 1 then it's ok, else we don't want to get a value
-            if res == 1:
-                c = self.db.connector._execute(None, sql) # Execute query returning the comment value
-                res = self.db.connector._fetchone(c)[0] # Fetch the comment value
-                self.db.connector._close_cursor(c) # Close cursor
-                self.editCom.setText(res) # Set comment value
-        except:
-            self.editCom.setEnabled(False)
+        objClass = self.db.searchClass()
+        if objClass != "PGDatabase":
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_field_properties.py]: No provider specific code here!

File: python/plugins/db_manager/db_plugins/spatialite/connector.py [hunk 10]
```diff
@@ -574,16 +571,7 @@ def runVacuum(self):
...
-        self._execute(None, sql)
-
-        sql = u"SELECT InvalidateLayerStatistics(%s)" % (self.quoteId(table))
-        self._execute(None, sql)
-
-        sql = u"SELECT UpdateLayerStatistics(%s)" % (self.quoteId(table))
-        self._execute(None, sql)
-
-        self._commit()
-        return True
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/spatialite/connector.py]: Are these changes on purpose?
Looks unrelated

File: python/plugins/db_manager/db_plugins/plugin.py [hunk 11]
```diff
@@ -1096,22 +1096,7 @@ def definition(self):
         return txt
 
     def getComment(self):
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/plugin.py]: Perfect, that's what we need here!

File: python/plugins/db_manager/db_plugins/oracle/plugin.py [hunk 12]
```diff
@@ -263,6 +261,9 @@ def registerDatabaseActions(self, mainWindow):
         mainWindow.registerAction(action, QApplication.translate(
             "DBManagerPlugin", "&Table"), self.emptyTableActionSlot)
 
+    def searchClass(self):
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/oracle/plugin.py]: Remove all those searchClass methods. All logic should stay encapsulated here.

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 13]
```diff
@@ -511,6 +511,25 @@ def getTableFields(self, table):
         self._close_cursor(c)
         return res
 
+    def setField(self, fld, tablename, db):
+        if fld is None:
+            return
+        print (tablename)
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/postgis/connector.py]: Debugging leftover, remove

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 14]
```diff
@@ -511,6 +511,25 @@ def getTableFields(self, table):
         self._close_cursor(c)
         return res
 
+    def setField(self, fld, tablename, db):
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/postgis/connector.py]: Shouldn't this method be called `getComment(tablename, fieldname)`?

File: python/plugins/db_manager/dlg_field_properties.py [hunk 15]
```diff
@@ -42,37 +41,22 @@ def __init__(self, parent=None, fld=None, table=None, db=None):
 
         for item in self.db.connector.fieldTypes():
             self.cboType.addItem(item)
-        self.setField(self.fld)
 
-        self.buttonBox.accepted.connect(self.onOK)
 
-    def setField(self, fld):
```
Comment (m-kuhn) [inline python/plugins/db_manager/dlg_field_properties.py]: Leave this method and just call `getComment` on the connector instead.

**Comment by m-kuhn** [comment]
Thanks a lot for the update.
This is on a good way but will need some additional changes before it can be shipped.

File: python/plugins/db_manager/dlg_import_vector.py (hunk 2 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/dlg_import_vector.py]: Well i think we can drop it, it didn't add it personnaly, i think it's either from the build on QGIS or a missclick or something

File: python/plugins/db_manager/dlg_import_vector.py (hunk 4 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/dlg_import_vector.py]: Oh well yes that's a good idea

File: python/plugins/db_manager/db_plugins/oracle/plugin.py (hunk 12 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/db_plugins/oracle/plugin.py]: searchClass must be replaced by supportComment() rigth ?

Comment (m-kuhn) [reply python/plugins/db_manager/db_plugins/oracle/plugin.py]: yes

File: python/plugins/db_manager/dlg_import_vector.py (hunk 2 above)
Comment (Gustry) [reply python/plugins/db_manager/dlg_import_vector.py]: It's the result of a bad merge I think with the master branch. This has been removed 2 days ago in the master branch

**Comment by Ailurupoda** [comment]
I'll keep working on it. I've seen some commit that had nothing to do with the fix of the issue, so i'll remove them. Not sure on how hard i can work on this today

File: python/plugins/db_manager/dlg_import_vector.py (hunk 2 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/dlg_import_vector.py]: Oh okay ! Thanks . Anyway i will check all commit and confirm them if they are needed.

**Comment by Ailurupoda** [comment]
```
ERROR: testCreateRenameDeleteFields (__main__.TestPyQgsDBManagerSpatialite)

Traceback (most recent call last):
  File "/root/QGIS/build/output/python/plugins/db_manager/db_plugins/connector.py", line 86, in _execute
    cursor.execute(sql)
sqlite3.OperationalError: duplicate column name: real_field
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/root/QGIS/tests/src/python/test_db_manager_spatialite.py", line 243, in testCreateRenameDeleteFields
    self.assertTrue(table.addField(field))
  File "/root/QGIS/build/output/python/plugins/db_manager/db_plugins/plugin.py", line 750, in addField
    ret = self.database().connector.addTableColumn((self.schemaName(), self.name), fld.definition())
  File "/root/QGIS/build/output/python/plugins/db_manager/db_plugins/spatialite/connector.py", line 592, in addTableColumn
    self._execute(None, sql)
  File "/root/QGIS/build/output/python/plugins/db_manager/db_plugins/connector.py", line 89, in _execute
    raise ConnectionError(e)
plugins.db_manager.db_plugins.plugin.ConnectionError: duplicate column name: real_field

```

I've got an issue from travis that concern a function i didn't change (or at least i think i didn't).
The spatialite addTableField, i don't manage to understand why it says that i have a duplicate column name and where i can change that.
Does someone have any idea ?

File: python/plugins/db_manager/db_plugins/gpkg/connector.py [hunk 16]
```diff
@@ -620,6 +620,20 @@ def runVacuum(self):
         """ run vacuum on the db """
         self._execute_and_commit("VACUUM")
 
+    def commentTable(self, schema, tablename, comment):
+        """Comment the table"""
+        return ''
+
+    def commentTable(self, schema, tablename, comment):
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/gpkg/connector.py]: This method is duplicated ,on purpose?

File: python/plugins/db_manager/db_plugins/gpkg/connector.py [hunk 17]
```diff
@@ -620,6 +620,20 @@ def runVacuum(self):
...
+        return ''
+
+    def unCommentTable(self, schema, tablename):
+        return ''
+
+    def getComment(self, tab, field, db):
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/gpkg/connector.py]:
```suggestion
    def getComment(self, tablename, field, db):
```
Be explicit ;)

File: python/plugins/db_manager/db_plugins/spatialite/connector.py [hunk 18]
```diff
@@ -571,18 +571,33 @@ def runVacuum(self):
...
+
     def addTableColumn(self, table, field_def):
         """ add a column to table """
         sql = u"ALTER TABLE %s ADD %s" % (self.quoteId(table), field_def)
+        self._execute_and_commit(sql)
         self._execute(None, sql)
 
-        sql = u"SELECT InvalidateLayerStatistics(%s)" % (self.quoteId(table))
-        self._execute(None, sql)
+        # sql = u"SELECT InvalidateLayerStatistics(%s)" % (self.quoteId(table))
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/spatialite/connector.py]: What's this for?

File: python/plugins/db_manager/db_plugins/spatialite/connector.py [hunk 19]
```diff
@@ -571,18 +571,33 @@ def runVacuum(self):
...
+        return ''
+
+    def getComment(self, tab, field, db):
+        """Returns the comment for a field"""
+        return ''
+
     def addTableColumn(self, table, field_def):
         """ add a column to table """
         sql = u"ALTER TABLE %s ADD %s" % (self.quoteId(table), field_def)
+        self._execute_and_commit(sql)
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/spatialite/connector.py]: What's this for?

File: python/plugins/db_manager/db_plugins/gpkg/connector.py (hunk 16 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/db_plugins/gpkg/connector.py]: Oupsy, not on purpose, didn't saw that :/ ty

File: python/plugins/db_manager/db_plugins/gpkg/connector.py [hunk 20]
```diff
@@ -620,6 +620,20 @@ def runVacuum(self):
...
         self._execute_and_commit("VACUUM")
 
+    def commentTable(self, schema, tablename, comment):
+        """Comment the table"""
+        return ''
+
+    def commentTable(self, schema, tablename, comment):
+        return ''
+
+    def unCommentTable(self, schema, tablename):
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/gpkg/connector.py]: I'd say a single `commentTable(self, schema, tablename, comment=None)` should be sufficient

File: python/plugins/db_manager/db_plugins/spatialite/connector.py (hunk 18 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/db_plugins/spatialite/connector.py]: I'm not the one writing it. It was dropped from the master because of a bad merge, so i just wrote it again

Comment (m-kuhn) [reply python/plugins/db_manager/db_plugins/spatialite/connector.py]: If it shows a difference in this pull request, it means that there's a difference between your branch and current master. Make sure they don't show up like this in here.

File: python/plugins/db_manager/db_plugins/spatialite/connector.py (hunk 19 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/db_plugins/spatialite/connector.py]: Same here, i wrote it again cause a bad merge deleted it . I don't know what it is for

Comment (m-kuhn) [reply python/plugins/db_manager/db_plugins/spatialite/connector.py]: Undo!

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 21]
```diff
@@ -742,18 +742,19 @@ def renameTable(self, table, new_table):
 
         self._commit()
 
-    def commentTable(self, schema, tablename, comment, db):
-        db.connector._execute(None, 'COMMENT ON TABLE "{0}"."{1}" IS E\'{2}\';'.format(schema, tablename, comment))
+    def commentTable(self, schema, tablename, comment=None, db):
+        if comment = None:
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/postgis/connector.py]: Almost sure you wanted to do a `==` here

File: python/plugins/db_manager/db_plugins/spatialite/connector.py (hunk 18 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/db_plugins/spatialite/connector.py]: Aren't the difference related to the last commit of the pull request ?

File: python/plugins/db_manager/db_plugins/spatialite/connector.py (hunk 19 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/db_plugins/spatialite/connector.py]: Well in fact i don't know where it comes from.. Dropped

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 22]
```diff
@@ -742,6 +742,29 @@ def renameTable(self, table, new_table):
 
         self._commit()
 
+    def commentTable(self, schema, tablename, db, comment=None):
+        if comment == None:
```
Comment (roya0045) [inline python/plugins/db_manager/db_plugins/postgis/connector.py]: Wouldn't it be better to go for `comment is None` rather than ==?

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 23]
```diff
@@ -742,6 +742,29 @@ def renameTable(self, table, new_table):
 
         self._commit()
 
+    def commentTable(self, schema, tablename, db, comment=None):
```
Comment (m-kuhn) [inline python/plugins/db_manager/db_plugins/postgis/connector.py]: kill the `db` parameter
Instead us

`self._execute(...)` below

**Comment by m-kuhn** [comment]
Cool, good job, this looks reasonably good to me now :+1: 
@elpaso can you have a look too?

File: python/plugins/db_manager/db_plugins/gpkg/connector.py [hunk 24]
```diff
@@ -620,6 +620,14 @@ def runVacuum(self):
         """ run vacuum on the db """
         self._execute_and_commit("VACUUM")
 
+    def commentTable(self, schema, tablename, comment=None):
+        """Comment the table"""
+        return ''
+
+    def getComment(self, tablename, field):
```
Comment (elpaso) [inline python/plugins/db_manager/db_plugins/gpkg/connector.py]: Move this method `getComment` into the base class `DBConnector` as a default implementation and override in PG connector only, instead of providing the same default implementation in all connectors but the PG one.

File: python/plugins/db_manager/db_plugins/oracle/connector.py [hunk 25]
```diff
@@ -1307,6 +1307,14 @@ def renameSchema(self, schema, new_schema):
         # Unsupported in Oracle
         pass
 
+    def commentTable(self, schema, tablename, comment=None):
+        """Comment the table"""
+        return ''
+
+    def getComment(self, tablename, field):
```
Comment (elpaso) [inline python/plugins/db_manager/db_plugins/oracle/connector.py]: Move this method `getComment` into the base class `DBConnector` as a default implementation and override in PG connector only, instead of providing the same default implementation in all connectors but the PG one.

File: python/plugins/db_manager/db_plugins/spatialite/connector.py [hunk 26]
```diff
@@ -571,6 +571,14 @@ def runVacuum(self):
         c.execute('VACUUM')
         self.connection.isolation_level = '' # reset to default isolation
 
+    def commentTable(self, schema, tablename, comment=None):
+        """Comment the table"""
+        return ''
+
+    def getComment(self, tablename, field):
```
Comment (elpaso) [inline python/plugins/db_manager/db_plugins/spatialite/connector.py]: Move this method `getComment` into the base class `DBConnector` as a default implementation and override in PG connector only, instead of providing the same default implementation in all connectors but the PG one.

File: python/plugins/db_manager/db_plugins/spatialite/plugin.py [hunk 27]
```diff
@@ -175,6 +175,9 @@ def explicitSpatialIndex(self):
     def spatialIndexClause(self, src_table, src_column, dest_table, dest_column):
         return u""" "%s".ROWID IN (\nSELECT ROWID FROM SpatialIndex WHERE f_table_name='%s' AND search_frame="%s"."%s") """ % (src_table, src_table, dest_table, dest_column)
 
+    def supportsComment(self):
```
Comment (elpaso) [inline python/plugins/db_manager/db_plugins/spatialite/plugin.py]: Move this method `getComment` into the base class `DBConnector` as a default implementation and override in PG connector only, instead of providing the same default implementation in all connectors but the PG one.

File: python/plugins/db_manager/db_plugins/vlayers/connector.py [hunk 28]
```diff
@@ -342,14 +342,22 @@ def runVacuum(self):
         print("**unimplemented** runVacuum")
         return False
 
+    def commentTable(self, schema, tablename, comment=None):
+        """Comment the table"""
+        return ''
+
+    def getComment(self, tablename, field):
```
Comment (elpaso) [inline python/plugins/db_manager/db_plugins/vlayers/connector.py]: Move this method `getComment` into the base class `DBConnector` as a default implementation and override in PG connector only, instead of providing the same default implementation in all connectors but the PG one.

File: python/plugins/db_manager/dlg_field_properties.py [hunk 29]
```diff
@@ -42,11 +41,18 @@ def __init__(self, parent=None, fld=None, table=None, db=None):
...
+        if supportCom != True:
+            self.label_6.setVisible(False)
+            self.editCom.setVisible(False)
+
+        self.setField(fld)
 
         self.buttonBox.accepted.connect(self.onOK)
 
     def setField(self, fld):
+        print('ok')
```
Comment (elpaso) [inline python/plugins/db_manager/dlg_field_properties.py]: Remove

File: python/plugins/db_manager/dlg_field_properties.py [hunk 30]
```diff
@@ -56,23 +62,16 @@ def setField(self, fld):
...
+        print(self.table)
+        tab = self.table.name
+        print(tab)
+        field = fld.name
+        print(field)
+        res = self.db.connector.getComment(tab, field)
+        print(res)
+        self.editCom.setText(res) # Set comment value
+        #except:
+        #    self.editCom.setEnabled(False)
```
Comment (elpaso) [inline python/plugins/db_manager/dlg_field_properties.py]: Remove all `print` instructions and the commented code

File: python/plugins/db_manager/dlg_table_properties.py [hunk 31]
```diff
@@ -50,6 +50,10 @@ def __init__(self, table, parent=None):
 
         self.db = self.table.database()
 
+        supportCom = self.db.supportsComment()
+        if supportCom != True:
```
Comment (elpaso) [inline python/plugins/db_manager/dlg_table_properties.py]:
```suggestion
        if not supportCom:
```

File: python/plugins/db_manager/db_plugins/gpkg/connector.py [hunk 32]
```diff
@@ -620,6 +620,14 @@ def runVacuum(self):
         """ run vacuum on the db """
         self._execute_and_commit("VACUUM")
 
+    def commentTable(self, schema, tablename, comment=None):
```
Comment (elpaso) [inline python/plugins/db_manager/db_plugins/gpkg/connector.py]: Move this method `commentTable` into the base class `DBConnector` as a default implementation and override in PG connector only, instead of providing the same default implementation in all connectors but the PG one.

File: python/plugins/db_manager/db_plugins/gpkg/connector.py [hunk 33]
```diff
@@ -620,6 +620,14 @@ def runVacuum(self):
...
+        """Comment the table"""
+        return ''
```
Comment (elpaso) [inline python/plugins/db_manager/db_plugins/gpkg/connector.py]: No need to return
```suggestion
        pass
```
but, IMO it would be cleaner if you checked for the connector to support comments before calling `commentTable` in `/python/plugins/db_manager/dlg_table_properties.py` instead of silently failing, it would be much more clear for whom read your caller code.

File: python/plugins/db_manager/dlg_table_properties.py [hunk 34]
```diff
@@ -334,36 +337,27 @@ def deleteIndex(self):
 
     def createComment(self):
         """Adds a comment to the selected table"""
-
         try:
-            #Using the db connector, executing de SQL query Comment on table
-            self.db.connector._execute(None, 'COMMENT ON TABLE "{0}"."{1}" IS E\'{2}\';'.format(self.table.schema().name, self.table.name, self.viewComment.text()))
+            schem = self.table.schema().name
```
Comment (elpaso) [inline python/plugins/db_manager/dlg_table_properties.py]: Should not you check here if comments are supported?

File: python/plugins/db_manager/dlg_table_properties.py [hunk 35]
```diff
@@ -334,36 +337,27 @@ def deleteIndex(self):
...
         #Display successful message
         QMessageBox.information(self, self.tr("Add comment"), self.tr("Table successfully commented"))
 
     def deleteComment(self):
         """Drops the comment on the selected table"""
-
         try:
-            #Using the db connector, executing de SQL query Comment on table using the NULL definition
-            self.db.connector._execute(None, 'COMMENT ON TABLE "{0}"."{1}" IS NULL;'.format(self.table.schema().name, self.table.name))
+            schem = self.table.schema().name
```
Comment (elpaso) [inline python/plugins/db_manager/dlg_table_properties.py]: ... and here too

**Review by elpaso — CHANGES_REQUESTED** [review]
Thanks for the patches! 
Please have a look to my comments

File: python/plugins/db_manager/dlg_table_properties.py (hunk 34 above)
Comment (Ailurupoda) [reply python/plugins/db_manager/dlg_table_properties.py]: Well in fact the method run only if you click the button "add comment" is pressed.
The button is up only if it supports comment. 
So for me i don't need to add an other check in the method, but if you think it's better i can.

**Review by elpaso — APPROVED** [review]

**Comment by elpaso** [comment]
@Ailurupoda thanks for your patience :+1:

**Comment by Ailurupoda** [comment]
@elpaso @m-kuhn  Thanks for your feed back !
Always better to do things a good way :+1:
